src/actions/posts.js

```diff
@@ -888,6 +890,93 @@ export function getOpenGraphMetadata(url) {
     };
 }
 
+export function getFlaggedPosts() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const userId = getCurrentUserId(state);
+        const teamId = getCurrentTeamId(state);
+
+        dispatch({type: SearchTypes.SEARCH_POSTS_REQUEST});
+        try {
+            const data = await Client4.getFlaggedPosts(userId, '', teamId);
+            dispatch(batchActions([
+                {type: SearchTypes.SEARCH_POSTS_SUCCESS},
+                {
+                    type: SearchTypes.RECEIVED_SEARCH_TERM,
+                    data: {
+                        term: null,
+                        do_search: false,
+                        is_mention_search: false
+                    }
+                },
+                {
+                    data,
+                    type: SearchTypes.RECEIVED_FLAGGED_POSTS
+                },
+                {
+                    data,
+                    type: SearchTypes.RECEIVED_SEARCH_POSTS
+                }
+            ]));
+
+            getProfilesAndStatusesForPosts(data.posts, dispatch, getState);
+            return data;
```

Comment:
Looks good, but can we move the dispatch, other actions and return out of the try block so it matches other actions?

Also we switched to using `return {data}` and `return {error}` [per this post](https://pre-release.mattermost.com/core/pl/ap1ue3xtobb6uxssdypq1rudyr), so if you could update the returns for this action that'd be great